src/python/pants/backend/python/dependency_inference/rules.py

```diff
@@ -89,6 +103,24 @@ class PythonInferSubsystem(Subsystem):
             "treated as a potential dependency if this option is set to 2 but not if set to 3."
         ),
     )
+    assets = BoolOption(
+        "--assets",
+        default=False,
+        help=(
+            "Infer a target's asset dependencies based on strings that look like Posix "
+            "filepaths, such as those given to `open` or `pkgutil.get_data`. To ignore any "
+            "false positives, put `!{bad_address}` in the `dependencies` field of your target."
+        ),
+    )
+    assets_min_slashes = IntOption(
+        "--assets-min-slashes",
```

Comment:
I'm thinking about imports in the form `pkgutil.get_data("path.to.module", "foo.txt")` and trying to figure out how we will want to model the options. It's fine even if we never implement it, but I suspect we may want to.

We'd probably want `--assets-min-dots`? I think it's plausible you'd want to have a different value for than that slashes? 

I was wondering if we could throw this into `--string-imports-min-dots`, but I don't think these are string imports. We need to preserve the file extension of `foo.txt`, whereas string imports are _Python_ imports.

Reply:
So this is getting into implementation details of a future change, so I wouldn't worry too much here.

The way I see it is that `pkgutil.get_data("path.to.module", "foo.txt")` is unambiguous, so wouldn't be subject to `min-dots` or `min-slashes` (but probably still subject to `assets`).

Reply:
> So this is getting into implementation details of a future change, so I wouldn't worry too much here.

I'm worried about it only so far as figuring out what option we likely want to have. It'd be great to avoid having to deprecate an option in the future.

--

The idea with min-slashes & min-dots is to help counteract false positives btw. We saw with pantsbuild.pants that setting too low caused way too many things to get inferred. 

--

I think we probably do want `--assets-min-dots`..If that's the case, then we're good to go. No need to think more about that on this PR.

Reply:
FWIW I think with some of the improvements (strong v weak imports and resolves) that technically those options aren't needed as much since the false positives are minimized even when running on all strings. But not everyone is using the right mix of options. (Thats for imports, at least)

In this case (and imports via function) I see the functional dependency (using the Pythonic function) as unambiguous. If you're not loading an asset via `pkgutil.get_data` what else are you doing?

Also, I don't really see where `--asset-min-dots` fits in :/ The only time we'd infer an asset and need dots is in `pkgutil.get_data` which is unambigous.

Reply:
From some DMs, there's some confusion here still about what I'm trying to say. 

But it's okay, I've thought it more and I think we would want two options. Meaning, no changes necessary to this option in this PR. We're good.